src/Component/LayerPreview/LayerPreview.tsx

```diff
@@ -0,0 +1,255 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState
+} from 'react';
+
+import {
+  useTranslation
+} from 'react-i18next';
+
+import OlMap from 'ol/Map';
+import OlLayer from 'ol/layer/Layer';
+import OlLayerTile from 'ol/layer/Tile';
+import OlLayerImage from 'ol/layer/Image';
+import OlSourceOSM from 'ol/source/OSM';
+import OlSourceTileWMS from 'ol/source/TileWMS';
+import OlSourceImageWMS from 'ol/source/ImageWMS';
+import OlSourceWMTS from 'ol/source/WMTS';
+import OlSourceVector from 'ol/source/Vector';
+import OlView from 'ol/View';
+import {
+  transformExtent
+} from 'ol/proj';
+
+import SHOGunApplicationUtil from '@terrestris/shogun-util/dist/parser/SHOGunApplicationUtil';
+import Layer from '@terrestris/shogun-util/dist/model/Layer';
+import {
+  getBearerTokenHeader
+} from '@terrestris/shogun-util/dist/security/getBearerTokenHeader';
+
+import LayerUtil from '@terrestris/ol-util/dist/LayerUtil/LayerUtil';
+
+import Logger from 'js-logger';
+
+import {
+  Alert,
+  Modal,
+  ModalProps,
+  Spin,
+  Tooltip
+} from 'antd';
+
+import {
+  GlobalOutlined
+} from '@ant-design/icons';
+
+import useSHOGunAPIClient from '../../Hooks/useSHOGunAPIClient';
+
+import MapComponent from './MapComponent/MapComponent';
+
+import './LayerPreview.less';
+
+function isWmsLayer(layer: OlLayer): layer is OlLayerImage<OlSourceImageWMS> | OlLayerTile<OlSourceTileWMS> {
+  if (layer instanceof OlLayer) {
+    const source = layer.getSource();
+    return source instanceof OlSourceImageWMS || source instanceof OlSourceTileWMS;
+  }
+  return false;
+}
+
+export interface LayerPreviewProps extends ModalProps {
+  layer: Layer;
+};
+
+export const LayerPreview: React.FC<LayerPreviewProps> = ({
+  layer,
+  ...passThroughProps
+}) => {
+  const [loading, setLoading] = useState(false);
+  const [addLayerError, setAddLayerError] = useState<string>();
+  const [extentError, setExtentError] = useState<string>();
+  const [loadLayerError, setLoadLayerError] = useState<string>();
+  const [open, setOpen] = useState(false);
+
+  const { t } = useTranslation();
+
+  const client = useSHOGunAPIClient();
+
+  const parser = useMemo(() => {
+    return new SHOGunApplicationUtil({
+      client: client
+    });
+  }, [client]);
+
+  const map = useMemo(() => {
+    return new OlMap({
+      layers: [
+        new OlLayerTile({
+          source: new OlSourceOSM()
+        })
+      ],
+      target: '',
+      view: new OlView({
+        center: [0, 0],
+        zoom: 0
+      })
+    });
+  }, []);
+
+  const addLayer = useCallback(async () => {
+    if (!layer) {
+      return;
+    }
+
+    try {
+      const olLayer = await parser.parseLayer(layer);
+      const layers = map.getAllLayers();
+      layers
+        .filter(l => l.get('shogunId') === layer.id)
+        .forEach(l => map.removeLayer(l));
+
+      const source = olLayer.getSource();
+
+      if (source instanceof OlSourceImageWMS) {
```

Comment:
```suggestion
      if (source instanceof OlSourceImage) {
```
This could also test for `OlSourceImage` , but I don't I know if that is supported or planned to be supported.

Reply:
Actually `ImageWMS` is the only supported one, but I agree while enabling a broader support here.

Reply:
See f9c348f.